### ISSUE TITLE
"Out of memory: Killed process" on Raspberry Pi

### ISSUE THREAD

**Issue by dfg3hz5zqtwz46hw6** [monero-project/monero#6425]
Hello,

monerod is being closed frequently on my Raspberry Pi 3B+
I'm running Ubuntu 18.04
I reckon it is related to #4189

```
root@ubuntu:/var/log# grep -a 'monero' ./syslog
Apr  3 11:09:13 ubuntu kernel: [44783.754209] monerod invoked oom-killer: gfp_mask=0x100cca(GFP_HIGHUSER_MOVABLE), order=0, oom_score_adj=0
Apr  3 11:09:13 ubuntu kernel: [44783.754233] CPU: 1 PID: 3242 Comm: monerod Tainted: G         C  E     5.3.0-1021-raspi2 #23~18.04.1-Ubuntu
Apr  3 11:09:13 ubuntu kernel: [44783.755802] [   3238]     0  3238   245719   137010  1753088        0             0 monerod
Apr  3 11:09:13 ubuntu kernel: [44783.755878] oom-kill:constraint=CONSTRAINT_NONE,nodemask=(null),cpuset=/,mems_allowed=0,global_oom,task_memcg=/,task=monerod,pid=3238,uid=0
Apr  3 11:09:13 ubuntu kernel: [44783.756235] Out of memory: Killed process 3238 (monerod) total-vm:982876kB, anon-rss:548040kB, file-rss:0kB, shmem-rss:0kB
Apr  3 11:09:13 ubuntu kernel: [44784.105520] oom_reaper: reaped process 3238 (monerod), now anon-rss:8kB, file-rss:0kB, shmem-rss:0kB
```

**Comment by sumogr** [comment]
As far as i can remember there was a bug with oom killer going rogue on xenial with early kernels. Can you update your kernel to the latest available version and see if it persists?

**Comment by dfg3hz5zqtwz46hw6** [comment]
Everything is up to date

**Comment by moneromooo-monero** [comment]
Do you have some reason to think this is a monero bug ? If so, what are they ? You mention nothing of that in your report.

**Comment by dfg3hz5zqtwz46hw6** [comment]
Because I am on a clean install of Ubuntu and only monerod crashes while everything else is running fine? Crashing is probably not intended behavior?


![htop](https://user-images.githubusercontent.com/21203981/78455435-b31aa680-769e-11ea-94e7-f276652c9262.PNG)

**Comment by sumogr** [comment]
on a pi with that setting on out and in peers what did you expect.
Reduce it

**Comment by fuwa0529** [comment]
add some swap

**Comment by dfg3hz5zqtwz46hw6** [comment]
What settings would you recommend?
How large should I make the swapfile? If I increase its size can I set more in and out peers?

**Comment by sumogr** [comment]
yes you should be able to. play a bit with it allocate 1 Gb of swap and reduce it to 50 and 50. Why is it so important that you exceed the default settings of incoming and outgoing nodes number?

**Comment by dfg3hz5zqtwz46hw6** [comment]
Thanks for the suggestions. I'll try it and will report back tomorrow if it doesn't crash again :)

I have a lot of bandwidth and I want to share as much as possible. Is there a better way to do it?

**Comment by moneromooo-monero** [comment]
Your log does not show a crash. It shows the OOM killer killing monerod.
This means your OS is short of RAM is decided to kill monerod to get some (because monerod uses a fair bit of it).

**Comment by dfg3hz5zqtwz46hw6** [comment]
I guess that's true. But there should be some kind of countermeasures in place for this behavior?
Anyway, I found a workaround for now.
I added 4Gb swap and am running monerod and the xmrig-proxy as systemctl services now.

[Unit]
Description=Monerod
After=network.target
StartLimitIntervalSec=0

[Service]
Restart=always
RestartSec=30
User=root
Nice=19
ExecStart=/usr/bin/screen -DmS monero /home/ubuntu/monero/monerod --limit-rate 50 --in-peers 50 --out-peers 50

[Install]
WantedBy=multi-user.target

Alternatively the OOM killer could be disabled I guess.
